devito/types/basic.py

```diff
@@ -932,7 +932,7 @@ def indexify(self, indices=None):
                     if s.free_symbols.intersection(self.args[i].free_symbols)]
 
         # Substitution for each index
-        subs = {s: 1 for s in spacings}
+        subs = {a*s: a for s in spacings for a in [-1, 1]}
```

Comment:
this dictionary will be reconstructed zilions of times in non-trivial pdes -- once for each Indexed. And in 3D for a TimeFunction, it's 8 sympy evaluations per Indexed. So I'm again a little concerned about speed. In my timings, `indexify` was indeed taking up some time. Can we build it once and for all ? 

```
@cached_property
def _spacing_subs(self):
    spacings = [i.spacing for i in self.dimensions]
    subs = {a*s: a for s in spacings for a in [-1, 1]}
    return subs
```
?

oc, to be retrieved from the mother function, ie `self.function._spacing_subs`

and then what is left to do in indexify are the filtering (`# Only keep the ones used as indices.`) and the actual substitutions

Reply:
I think any feedback welcome. I appreciate this is a minor and it should just go in and full stop, but many minor incautious patches will eventually kill off the compilation speeds improvements for which I spilled so much blood

Reply:
All of this was actually almost never called because there was some duplicate of it in `operator`.